src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -491,12 +492,14 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             CancellationTokenSource cts;
             bool disposeCts;
             bool hasTimeout = _timeout != s_infiniteTimeout;
+            long timeoutTime = 0;
             if (hasTimeout || cancellationToken.CanBeCanceled)
             {
                 disposeCts = true;
                 cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _pendingRequestsCts.Token);
                 if (hasTimeout)
                 {
+                    timeoutTime = Environment.TickCount64 + (long) _timeout.TotalMilliseconds;
```

Comment:
Can you please run some perf tests to see if this Environment.TickCount64 call shows up at all?

Reply:
Will do.

Reply:
I compared performance of a success GetAsync path without cancellation in the 2 following cases and found that TickCount64 access adds 1.84% to execution time.
1. Normal code reading Environment.TickCount64
```C# timeoutTime = Environment.TickCount64 + (long)_timeout.TotalMilliseconds;```
2. Hardcoded constant is used instead of Environment.TickCount64
```C# timeoutTime = 9223372000000000000L + (long)_timeout.TotalMilliseconds;```

Test code
```C#
            [GlobalSetup]
            public void Setup()
            {
                var response = new HttpResponseMessage(System.Net.HttpStatusCode.OK);
                client = new HttpClient(new CustomResponseHandler((r, c) => 
                    Task.FromResult(response)));
                client.Timeout = TimeSpan.FromDays(1);
                uri = new Uri($"http://{Guid.NewGuid().ToString("N")}");
            }

            [Benchmark]
            public void Run()
            {
                for (int i = 0; i < 10000; i++)
                {
                    client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead, 
                        cts.Token).Wait();
                }
            }
```

| Method |                                                                                                   Toolchain |     Mean |     Error |    StdDev | Ratio |
|------- |------------------------------------------------------------------------------------------------------------ |---------:|----------:|----------:|------:|
|    Run | **Constant long value** | **3.369** ms | 0.0189 ms | 0.0168 ms |  0.98 |
|    Run |  **Environment.TickCount64** | **3.431** ms | 0.0380 ms | 0.0317 ms |  1.00 |

Reply:
@stephentoub I'm curious -- I believe `GetTickCount64` is just `mov rax, [addr]` last I looked; is there something special in our implementation of `TickCount64` that is worth watching out for?

Reply:
I'm not following... are you referring to the mov that's part of the tail call to that function? The single mov isn't the entirety of TickCount64.

Reply:
I may be remembering incorrectly here; last time I looked was in XP x64 days. Regardless I don't think it's relevant to PR so convo so I'll go look independently 😊